### PR TITLE
add users routes and partials

### PR DESCRIPTION
**Pull request by CodeX-y**

### CODE REVIEW

**Review by CarlosCastanedaDev — APPROVED** [review]
Nice job, Yusra!

Great use of partials in your views!

**Comment by CarlosCastanedaDev** [comment]
Please submit your PR link in Canvas so I can grade it and you can move forward.

File: app/controllers/users_controller.rb
Comment (SaraDawner2000) [inline app/controllers/users_controller.rb:1]: You can DRY up the code a little by abstracting fetching the user into a private method and then calling it for all other methods with "before_action"

File: app/views/users/_navbar.html.erb
Comment (SaraDawner2000) [inline app/views/users/_navbar.html.erb:1]: great job moving the navbar to a partial!

File: app/views/users/feed.html.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<%= render partial: "users/navbar"%>
+
+<% @users.feed.each do |post| %>
```
Comment (SaraDawner2000) [inline app/views/users/feed.html.erb]: there is a typo: this should be @user.feed. The feed tab doesn't work

File: app/views/users/_users.html.erb
Comment (SaraDawner2000) [inline app/views/users/_users.html.erb:1]: You could use the "link_to" helper method here

File: app/views/users/followers.html.erb
Comment (SaraDawner2000) [inline app/views/users/followers.html.erb:1]: The followers tab doesn't load. Your list item partial is named _users.html.erb, in plural. The syntax for rendering it doesn't work. What I found worked was <%= render "users/users", user: user %>

File: app/views/users/following.html.erb
Comment (SaraDawner2000) [inline app/views/users/following.html.erb:1]: Same as above

**Review by SaraDawner2000 — CHANGES_REQUESTED** [review]
I like how you put everything into partials to improve readibility and save time! There are just a couple of issues with it currently, but they shouldn't be hard to fix

File: app/views/users/following.html.erb
Comment (CodeX-y) [reply app/views/users/following.html.erb:1]: Thank you so much! I made all the changes you found

**Pull request merged**